PlayableEmbeds/src/main/kotlin/dev/vendicated/aliucordplugins/betterplatformembeds/PlayableEmbeds.kt

```diff
@@ -50,25 +48,67 @@ class PlayableEmbeds : Plugin() {
                 "(?:embed/|v/|watch\\?v=|watch\\?.+&v=|shorts/))((\\w|-){11})"+
                 "(?:(?:\\?|&)(?:star)?t=(\\d+))?(?:\\S+)?")
 
-    override fun start(_context: Context) {
+    override fun start(context: Context) {
         patcher.after<WidgetChatListAdapterItemEmbed>("configureUI", WidgetChatListAdapterItemEmbed.Model::class.java) {
             val model = it.args[0] as WidgetChatListAdapterItemEmbed.Model
-            val embed = model.embedEntry.embed
+            val embed = MessageEmbedWrapper(model.embedEntry.embed)
             val holder = it.thisObject as WidgetChatListAdapterItemEmbed
             val layout = holder.itemView as ConstraintLayout
 
             layout.findViewById<WebView?>(widgetId)?.let { v ->
                 if (webviewMap[v] == embed.url) return@after
                 (v.parent as ViewGroup).removeView(v)
             }
-            val url = embed.url ?: return@after;
-            when (embed.rawProvider?.name) {
+            val url = embed.url ?: return@after
+            when (embed.provider?.name) {
                 "YouTube" -> addYoutubeEmbed(layout, url)
                 "Spotify" -> addSpotifyEmbed(layout, url)
+                else -> addDefaultEmbed(layout, embed)
             }
         }
     }
 
+    private fun addDefaultEmbed(layout: ViewGroup, embed: MessageEmbedWrapper) {
+        val ctx = layout.context
+
+        val videoUrl = embed.video?.url ?: return
+
+        val cardView = layout.findViewById<CardView>(Utils.getResId("embed_image_container", "id"))
+        val chatListItemEmbedImage = cardView.findViewById<SimpleDraweeView>(Utils.getResId("chat_list_item_embed_image", "id"))
+        val playButton = cardView.findViewById<View>(Utils.getResId("chat_list_item_embed_image_icons", "id"))
+        playButton.visibility = View.GONE
+        chatListItemEmbedImage.visibility = View.GONE
+        val posterUrl = embed.image?.url
```

Comment:
double check that this is discord media proxy url and not the external url, otherwise this can be used to ip log

Reply:
When I use the ProxyUrl, the videos don't load, that's why I wasn't using them. I don't know what's wrong with it

Reply:
The thumbnail can be proxied however. I'll make sure to enable that.